lib/shared/bucketing-assembly-script/assembly/bucketing/bucketing.ts

```diff
@@ -381,3 +385,35 @@ export function _generateBucketedVariableForUser(
     )
     return { variable: sdkVar, variation, feature: featureForVariable }
 }
+
+export function _getUserValueForBucketingKey(
+    user: DVCPopulatedUser,
+    target: PublicTarget
+): string {
+    if (target.bucketingKey && target.bucketingKey !== 'user_id') {
+        let bucketingValue: string = DEFAULT_BUCKETING_VALUE
+        const customData = user.customData
+        const privateCustomData = user.privateCustomData
+        if (customData) {
+            const customDataValue = getValueFromJSONOptional(customData, target.bucketingKey)
+            bucketingValue = customDataValue
+                ? customDataValue.toString()
+                : DEFAULT_BUCKETING_VALUE
+        } else if (privateCustomData) {
```

Comment:
should this not be checked when customDataValue is missing?

Reply:
The `User` class has a `getCombinedCustomData` function that should allow you to abstract away the private vs public custom data

Reply:
much better ty